### PR TITLE
Fix dead lock of terminal

### PR DESCRIPTION
**Pull request by Yaraslaut**
Fixes bug introduced in https://github.com/contour-terminal/contour/pull/1450

### CODE REVIEW

File: src/vtbackend/Terminal.h [hunk 1]
```diff
@@ -391,17 +391,9 @@ class Terminal
     void updateInputMethodPreeditString(std::string preeditString);
     // }}}
```
Comment (christianparpart) [inline src/vtbackend/Terminal.h:393]: Looks like we want to get rid of the inner lock then

Comment (Yaraslaut) [reply src/vtbackend/Terminal.h:393]: Deleted _innerLock

File: src/vtbackend/Terminal.cpp [hunk 2]
```diff
@@ -1274,8 +1272,6 @@ string Terminal::extractSelectionText() const
 
 string Terminal::extractLastMarkRange() const
 {
-    auto const _ = std::lock_guard { *this };
```
Comment (christianparpart) [inline src/vtbackend/Terminal.cpp:1277]: if we remove this, we need to apply the lock from outside, i.e. in

```cpp
bool TerminalSession::operator()(actions::CopyPreviousMarkRange)
```

File: src/vtbackend/Terminal.cpp [hunk 3]
```diff
@@ -1253,8 +1253,6 @@ namespace
 
 string Terminal::extractSelectionText() const
 {
-    auto const _ = std::scoped_lock { *this };
```
Comment (christianparpart) [inline src/vtbackend/Terminal.cpp:1256]: Wen removed here, we need to add one from outside here:

```cpp
bool TerminalSession::operator()(actions::CopySelection copySelection);
bool TerminalSession::operator()(actions::OpenSelection);
```

All the other cases I found were indirectly governed by the caller (/of a caller) already.

**Review by christianparpart — APPROVED** [review]
LGTM. Many thanks, @Yaraslaut

**Pull request merged**